src/java/org/apache/cassandra/schema/Schema.java

```diff
@@ -77,8 +77,19 @@ public class Schema implements SchemaProvider
 
     public static final Schema instance = new Schema();
 
-    private volatile Keyspaces distributedKeyspaces = Keyspaces.none();
-    private volatile Keyspaces distributedAndLocalKeyspaces;
+    private static class Holder
+    {
+        private final Keyspaces distributedKeyspaces;
+        private final Keyspaces distributedAndLocalKeyspaces;
+
+        private Holder(Keyspaces distributedKeyspaces, Keyspaces distributedAndLocalKeyspaces)
```

Comment:
This is just to force both to be updated at the same time and avoid them getting out of sync?

Reply:
yes, instead of making getter synchronized as it would be slower and lead to deadlocks.